src/Service/PaymentRestrictionValidation/BasePaymentRestrictionValidation.php

```diff
@@ -133,9 +142,14 @@ private function isCurrencySupportedByPaymentName($paymentName)
     {
         $enabledCurrencies = $this->getEnabledCurrenciesByPaymentName($paymentName);
 
-        $isAllCurrencies = in_array('0', $enabledCurrencies);
-        $isCurrencyInList = in_array($this->legacyContext->getCurrencyId(), $enabledCurrencies);
+        if (in_array('0', $enabledCurrencies)) {
+            $enabledCurrenciesIsoCode = $this->obtainPaymentMethods->obtainPaymentMethods()[$paymentName]['currencies'];
+            foreach ($enabledCurrenciesIsoCode as $isoCode) {
+                $currencyId = Currency::getIdByIsoCode($isoCode);
+                $enabledCurrencies[$currencyId] = $currencyId;
```

Comment:
Seems a bit weird to fill up same array :D

Reply:
if there is 0 then we fill it, so there is no issue leaving the 0 :D i guess i can reset the array but is not big deal